### PR TITLE
vdr: revisited version of https://github.com/NixOS/nixpkgs/pull/32050

### PR DESCRIPTION
**Pull request by Mic92**
###### Motivation for this change

- rebased version of https://github.com/NixOS/nixpkgs/pull/32050
- current status, stops after start:
● vdr.service - VDR
   Loaded: loaded (/nix/store/m1ncnwmkrwjr06k43psyh3xdr0i83qd7-unit-vdr.service/vdr.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2018-12-22 16:51:35 CET; 19min ago
  Process: 15152 ExecStart=/nix/store/7267sbr3ddqaqlmg97g6vgv4h59pppxr-vdr-2.4.0/bin/vdr --video=/srv/vdr/video --config=/var/lib/vdr (code=exited, status=2)
 Main PID: 15152 (code=exited, status=2)

Dec 22 16:51:35 turingmachine systemd[1]: vdr.service: Service RestartSec=100ms expired, scheduling restart.
Dec 22 16:51:35 turingmachine systemd[1]: vdr.service: Scheduled restart job, restart counter is at 5.
Dec 22 16:51:35 turingmachine systemd[1]: Stopped VDR.
Dec 22 16:51:35 turingmachine systemd[1]: vdr.service: Start request repeated too quickly.
Dec 22 16:51:35 turingmachine systemd[1]: vdr.service: Failed with result 'exit-code'.
Dec 22 16:51:35 turingmachine systemd[1]: Failed to start VDR.

@ck3d I squashed your commits and added systemd-tmpfile rules to automatically create the video directory on startup. However the services still fails, when
I add `services.vdr.enable = true;`. What else do I need? I do not own any video/dvb-t devices.

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build vdr

**Comment by ck3d** [comment]
First of all, I expect following error (should be visible in journal):
```
vdr: error while reading '/var/lib/vdr/sources.conf'
vdr: error while reading '/var/lib/vdr/channels.conf'
vdr: no primary device found - using first device!
```
A good default for sources.conf can be found in ${pkgs.vdr}/share/vdr/conf or http://www.vdr-wiki.de/wiki/index.php/Sources.conf .
Channels.conf depends on the DVB  source you want to use. If you have german DVB-S, you can use the channels.conf which is in the same folder as the sources.conf or you can start creating our own. I can recommend following page http://vdr-wiki.de/wiki/index.php/Channels.conf .

But the main issue is not having a DVB device. It seems that vdr doesn't start if no DVB device is available.

**Comment by ck3d** [comment]
/var/lib/vdr is the place where vdr also stores recording timers in timer.conf and user settings in setup.conf . Vdr can also update channel names and add new channels in channels.conf . All other files are more or less static, but it is not possible to set different config paths . For more information following page gives an good overview http://vdr-wiki.de/wiki/index.php/Struktur .

I got this PR on my media PC perfectly running. The tmprule looks good.

**Comment by ck3d** [comment]
@Mic92 Thanks a lot for helping! Great that you merged the PR!
I do not know why I did not see it yesterday, but today my recordings have been deleted! I think something is wired with the tmprules. We should remove the tmprules or fix is?!

File: nixos/modules/services/hardware/vdr.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        description = "Additional command line arguments to pass to VDR.";
+      };
+    };
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      "d ${cfg.videoDir} 0755 vdr vdr 0"
```
Comment (Mic92) [inline nixos/modules/services/hardware/vdr.nix]: Sorry this should be -:

```
"d ${cfg.videoDir} 0755 vdr vdr -"
```

**Comment by Mic92** [comment]
fixed in 044ff3dc663

**Comment by ck3d** [comment]
Thanks for the fast fix. Now I understand the 0. ... I hope I can recover my recordings. Nevertheless thanks a lot for bringing the PR into Nixpkgs.

**Pull request merged**